From 75efa3610068ee48f303dc0938a4ad92af8a8000 Mon Sep 17 00:00:00 2001
From: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
Date: Mon, 29 Apr 2024 15:43:48 +0530
Subject: [PATCH] drivers: bam_dma: devsoc: fix BAM_DESC_CNT_TRSHLD
 configuration

BAM_DESC_CNT_TRSHLD register is not present in devsoc because
it follows bam-lite architecture. So configure this register
only for non bam-lite soc's.

Change-Id: I50c4b72ef0a2f63051fa0f370d096aa526b5661f
Signed-off-by: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
---
 drivers/dma/qcom/bam_dma.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/qcom/bam_dma.c b/drivers/dma/qcom/bam_dma.c
index 8510cf1ba967..b0d4fc212444 100644
--- a/drivers/dma/qcom/bam_dma.c
+++ b/drivers/dma/qcom/bam_dma.c
@@ -400,6 +400,8 @@ struct bam_device {
 
 	/* dma start transaction tasklet */
 	struct tasklet_struct task;
+
+	bool is_bam_lite;
 };
 
 /**
@@ -443,8 +445,9 @@ static void bam_reset(struct bam_device *bdev)
 	writel_relaxed(val, bam_addr(bdev, 0, BAM_CTRL));
 
 	/* set descriptor threshhold, start with 4 bytes */
-	writel_relaxed(DEFAULT_CNT_THRSHLD,
-			bam_addr(bdev, 0, BAM_DESC_CNT_TRSHLD));
+	if (!bdev->is_bam_lite)
+		writel_relaxed(DEFAULT_CNT_THRSHLD,
+			       bam_addr(bdev, 0, BAM_DESC_CNT_TRSHLD));
 
 	/* Enable default set of h/w workarounds, ie all except BAM_FULL_PIPE */
 	writel_relaxed(BAM_CNFG_BITS_DEFAULT, bam_addr(bdev, 0, BAM_CNFG_BITS));
@@ -1307,6 +1310,9 @@ static int bam_dma_probe(struct platform_device *pdev)
 		return ret;
 	}
 
+	bdev->is_bam_lite = of_property_read_bool(pdev->dev.of_node,
+						  "qcom,bam-lite");
+
 	ret = bam_init(bdev);
 	if (ret)
 		goto err_disable_clk;
-- 
2.34.1

